### PR TITLE
Log message source details are grouped

### PR DESCRIPTION
**Pull request by majorosdonat**
Log message source details are grouped: they are not relevant for most users and can distract them from finding the root cause of their errors.

image

---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/contributing-docs/05_pull_requests.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Review by jscheffl — APPROVED** [review]
Thanks for the fix, looks good for me - also valid to back-port to 2.10 line in my view.

@bbovenzi would it be okay to back-port also for you?

We had a couple of complaints from users who had (1) error markers in logs in UI even marked in red just because the celery worker was not delivering the logs internally - but logs were loaded from file successfully and (2) if logs were archived on blob storage the internal blob storage URL was posted which actually is not accessible for users.

So the first lines posted in the logs mainly confused and were not helpful... Donat came to the proposal to hide/log group them per default.

**Comment by bbovenzi** [comment]
Makes sense to me!

File: providers/tests/amazon/aws/log/test_s3_task_handler.py [hunk 1]
```diff
@@ -128,7 +128,10 @@ def test_read(self):
         ti.state = TaskInstanceState.SUCCESS
         log, metadata = self.s3_task_handler.read(ti)
         actual = log[0][0][-1]
-        expected = "*** Found logs in s3:\n***   * s3://bucket/remote/log/location/1.log\nLog line"
+        expected = " INFO - ::group::Log message source details\n"
+        expected += "*** Found logs in s3:\n***   * s3://bucket/remote/log/location/1.log\n"
+        expected += " INFO - ::endgroup::\n"
+        expected += "Log line"
```
Comment (jscheffl) [inline providers/tests/amazon/aws/log/test_s3_task_handler.py]: Ah we had a bit of side-discussions about the exact match and... the Airflow CI now shows errors in back-compat tests.

Provider packages are tested for compatibility against old Airflow versions 2.10, 2.9 and 2.8. As the log grouping was not existing like this in the former Airflow-Versions, these tests fail...

So most probably, to make CI green you need to change the exact string match to an non-exact match like `assert "my expected text" in actual`
...then you are Airflow core version neutral.

Comment (jscheffl) [inline providers/tests/amazon/aws/log/test_s3_task_handler.py]: ...so for example like:
```suggestion
        assert "*** Found logs in s3:\n***   * s3://bucket/remote/log/location/1.log" in actual
        assert "Log line" in actual
```

File: providers/tests/amazon/aws/log/test_s3_task_handler.py [hunk 2]
```diff
@@ -129,7 +129,8 @@ def test_read(self):
         log, metadata = self.s3_task_handler.read(ti)
         actual = log[0][0][-1]
         expected = "*** Found logs in s3:\n***   * s3://bucket/remote/log/location/1.log\nLog line"
```
Comment (jscheffl) [inline providers/tests/amazon/aws/log/test_s3_task_handler.py]: Just a small glitch, then I assume it is good to go:
```
 providers/tests/amazon/aws/log/test_s3_task_handler.py:131:9: F841 Local variable `expected` is assigned to but never used
    |
129 |         log, metadata = self.s3_task_handler.read(ti)
130 |         actual = log[0][0][-1]
131 |         expected = "*** Found logs in s3:\n***   * s3://bucket/remote/log/location/1.log\nLog line"
    |         ^^^^^^^^ F841
132 |         assert "*** Found logs in s3:\n***   * s3://bucket/remote/log/location/1.log\n" in actual
133 |         assert actual.endswith("Log line")
    |
    = help: Remove assignment to unused variable `expected`
```

**Pull request merged**